official/resnet/resnet.py

```diff
@@ -254,8 +344,10 @@ def block_layer(inputs, filters, block_fn, blocks, strides, training, name,
   Returns:
     The output tensor of the block layer.
   """
+  filters_out = filters
   # Bottleneck blocks end with 4x the number of filters as they start with
-  filters_out = 4 * filters if block_fn is bottleneck_block else filters
+  if block_fn is _bottleneck_block_v1 or block_fn is _bottleneck_block_v2:
```

Comment:
This identity check is error-prone. Let's just pass in the bottleneck bool that we already have for this check.